### ISSUE TITLE
Import milli only once in Meilisearch

### ISSUE THREAD

**Issue by Kerollmops** [meilisearch/meilisearch#2265]
We currently import milli in _meilisearch-lib_ and _meilisearch-auth_ we should find a way to import it only once in the repository, it will help avoid importing two different versions and break builds or anything.

**Comment by lucas-cauhe** [comment]
This issue seems to have been present for a long while and finally been merged into [cargo's main branch](https://github.com/rust-lang/cargo) just a week ago!!

It was first tackled [here](https://github.com/rust-lang/cargo/issues/3931), later derived into this [RFC](https://github.com/rust-lang/rfcs/pull/2906) and finally was made a [PR](https://github.com/rust-lang/cargo/pull/10497). 
Now I guess you could either clone cargo's repo or wait for a new release (I don't really know whether there's some other option)

**Comment by curquiza** [comment]
@Kerollmops is it still true?
If I'm correct this issue is solved, milli is only present in meilisearch-types?

**Comment by Kerollmops** [comment]
Indeed, @lucas-cauhe is right. Every crate in the meilisearch git repository is importing the `meilisearch-types` crate which is itself importing milli. No other `milli` dependency is made in any other crate than this one. In conclusion, we only import `milli` once in the `meilisearch-types` crate.
